MAINT: runtests.py overrides $PYTHONPATH

Similar to scipy/scipy#14501, which was fixed in scipy/scipy#14928 (see https://github.com/scipy/scipy/pull/14928#issuecomment-953740050)

`runtest.py` currently overrides the `$PYTHONPATH` environment variable when setting up the environment for child scripts to be invoked. This leads child processes, e.g. `setup.py`, to not run in the same environment as the parent, i.e. they cannot find python modules installed under a separate prefix.

@melissawm I am having some problems with running `runtests.py` script. I think something wrong with dependencies and development environment.

Python version: Python 3.10.0
Environment: Docker

Output:
```
vscode ➜ /workspaces/numpy-andrei/numpy (issue#20239 ✗) $ pip install .
Defaulting to user installation because normal site-packages is not writeable
Processing /workspaces/numpy-andrei/numpy
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing wheel metadata (pyproject.toml) ... done
Building wheels for collected packages: numpy
  Building wheel for numpy (pyproject.toml) ... done
  Created wheel for numpy: filename=numpy-1.22.0.dev0+1605.g927feff87-cp310-cp310-linux_x86_64.whl size=22942752 sha256=31c6cfb7c2857c2560fd6f45b54e396e8ee03456b52d3df1fba8bbd2655a05c0
  Stored in directory: /tmp/pip-ephem-wheel-cache-6pz5i2lm/wheels/a0/fc/dc/30d0dcee460b182043616917c663cdddb3d468cc924782b6b7
Successfully built numpy
Installing collected packages: numpy
Successfully installed numpy-1.22.0.dev0+1605.g927feff87
WARNING: You are using pip version 21.3; however, version 21.3.1 is available.
You should consider upgrading via the '/usr/local/bin/python3 -m pip install --upgrade pip' command.
vscode ➜ /workspaces/numpy-andrei/numpy (issue#20239 ✗) $ python runtests.py -v -m full
Building, see build.log...
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
    ... build in progress
Build OK
/workspaces/numpy-andrei/numpy/build/testenv/lib/python3.10/site-packages/numpy/distutils/ccompiler.py:8: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  from distutils import ccompiler
/workspaces/numpy-andrei/numpy/build/testenv/lib/python3.10/site-packages/numpy/distutils/ccompiler.py:17: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
  from distutils.sysconfig import customize_compiler
NumPy version 1.22.0.dev0+1605.g927feff87
NumPy relaxed strides checking option: True
NumPy CPU features:  SSE SSE2 SSE3 SSSE3* SSE41* POPCNT* SSE42* AVX* F16C* FMA3* AVX2* AVX512F? AVX512CD? AVX512_KNL? AVX512_KNM? AVX512_SKX? AVX512_CLX? AVX512_CNL? AVX512_ICL?
ImportError while loading conftest '/workspaces/numpy-andrei/numpy/build/testenv/lib/python3.10/site-packages/numpy/conftest.py'.
numpy/conftest.py:7: in <module>
    import hypothesis
E   ModuleNotFoundError: No module named 'hypothesis'
vscode ➜ /workspaces/numpy-andrei/numpy (issue#20239 ✗) $ 
```

Could you help me to fix the issue.

--- a/runtests.py
+++ b/runtests.py
@@ -192,7 +192,12 @@ def main(argv):
         site_dir, site_dir_noarch = build_project(args)
         sys.path.insert(0, site_dir)
         sys.path.insert(0, site_dir_noarch)
-        os.environ['PYTHONPATH'] = site_dir + os.pathsep + site_dir_noarch
+        os.environ['PYTHONPATH'] = \
+            os.pathsep.join((
+                site_dir, 
+                site_dir_noarch, 
+                os.environ.get('PYTHONPATH', '')
+            ))
     else:
         _temp = __import__(PROJECT_MODULE)
         site_dir = os.path.sep.join(_temp.__file__.split(os.path.sep)[:-2])
@@ -493,7 +498,8 @@ def build_project(args):
         os.makedirs(site_dir)
     if not os.path.exists(site_dir_noarch):
         os.makedirs(site_dir_noarch)
-    env['PYTHONPATH'] = site_dir + os.pathsep + site_dir_noarch
+    env['PYTHONPATH'] = \
+        os.pathsep.join((site_dir, site_dir_noarch, env.get('PYTHONPATH', '')))
 
     log_filename = os.path.join(ROOT_DIR, 'build.log')
 

